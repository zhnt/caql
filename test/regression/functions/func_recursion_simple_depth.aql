// Recursion depth test cases (non-tail call)

// Sum function (non-tail recursive)
function sum_to_n(n) {
    if n <= 0 {
        return 0
    }
    return n + sum_to_n(n - 1)  // Non-tail recursive
}

print("=== Recursion Depth Tests ===")

let s15 = sum_to_n(15)
print("sum_to_n(15) = " + string(s15))