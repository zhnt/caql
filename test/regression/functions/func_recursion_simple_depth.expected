=== Recursion Depth Tests ===
sum_to_n(15) = 120
